test/test_torch.py

```diff
@@ -5921,36 +5923,33 @@ def run(device, data, model, optimizer, scaler, loss_fn, skip_iter, try_scaling_
             )
 
     @onlyNativeDeviceTypes
-    def test_grad_scaling_autocast(self, device):
-        device = torch.device(device)
-        for optimizer_ctor in (torch.optim.SGD, torch.optim.Adam, torch.optim.AdamW):
-            self._grad_scaling_autocast_test(device=device.type, optimizer_ctor=optimizer_ctor)
-
-    @onlyNativeDeviceTypes
-    def test_grad_scaling_autocast_foreach(self, device):
-        device = torch.device(device)
-        for optimizer_ctor in (torch.optim.SGD, torch.optim.Adam, torch.optim.AdamW):
-            self._grad_scaling_autocast_test(device=device.type, optimizer_ctor=optimizer_ctor, optimizer_kwargs={"foreach": True})
-
-    @onlyCUDA
-    def test_grad_scaling_autocast_fused(self, device):
+    @optims(
+        [optim for optim in optim_db if optim.optim_cls in [torch.optim.AdamW, torch.optim.Adam, torch.optim.SGD]],
+        dtypes=[torch.float32]
+    )
+    def test_grad_scaling_autocast(self, device, dtype, optim_info):
         device = torch.device(device)
-        for optimizer_ctor in (torch.optim.Adam, torch.optim.AdamW):
+        optimizer_ctor = optim_info.optim_cls
+        self._grad_scaling_autocast_test(device=device.type, optimizer_ctor=optimizer_ctor)
+        self._grad_scaling_autocast_test(device=device.type, optimizer_ctor=optimizer_ctor, optimizer_kwargs={"foreach": True})
+        if device.type == "cuda":
             self._grad_scaling_autocast_test(device=device.type, optimizer_ctor=optimizer_ctor, optimizer_kwargs={"fused": True})
```

Comment:
There's a way we can go even further here--I ultimately don't think we should need a `_grad_scaling_autocast_test`. `_grad_scaling_autocast_test` should just be `test_grad_scaling_autocast` and the different kwargs should be retrieved with `_get_optim_inputs_including_global_cliquey_kwargs` https://github.com/pytorch/pytorch/blob/main/test/test_optim.py#L501C32-L501C77 though you'd want to only skip differentiable.

This will increase coverage (at least 5x), but I don't think that's a bad thing!